### PR TITLE
Add a new global plugin API.

### PR DESCRIPTION
**Pull request by Laimiux**
## What
Enable global introspection via a plugins API.

### CODE REVIEW

**Comment by carrotkite** [comment]
### JaCoCo Code Coverage 77.44% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `com/instacart/formula/coroutines/FlowRuntime` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/internal/ListInspector` | 0% | 0% | :white_check_mark: |
| `com/instacart/formula/rxjava3/RxJavaRuntime` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/FormulaPlugins` | 33% | 0% | :white_check_mark: |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: formula/src/main/java/com/instacart/formula/internal/ListInspector.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.instacart.formula.Inspector
+import kotlin.reflect.KClass
+
+internal class ListInspector(
+    private val inspectors: List<Inspector>,
+): Inspector {
+    override fun onFormulaStarted(formulaType: KClass<*>) {
+        for (inspector in inspectors) {
+            inspector.onFormulaStarted(formulaType)
+        }
```
Comment (alexanderbezverhni) [inline formula/src/main/java/com/instacart/formula/internal/ListInspector.kt:13]: nit: maybe introduce `private fun forEveryInspector(block: Inspector.() -> Unit)` to make this class less verbose and a little more kt idiomatic?

```
forEveryInspector { onFormulaStarted(formulaType) }
```

File: formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt [hunk 2]
```diff
@@ -1249,33 +1254,41 @@ class FormulaRuntimeTest(val runtime: TestableRuntime, val name: String) {
 
     @Test
     fun `inspector events`() {
+        val globalInspector = TestInspector()
+        FormulaPlugins.setPlugin(object : Plugin {
+            override fun inspector(type: KClass<*>): Inspector {
+                return globalInspector
+            }
+        })
```
Comment (alexanderbezverhni) [inline formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt:1262]: nit: should we have test cases for any possible (global, local) permutation, not only when both are not null?

**Review by alexanderbezverhni — APPROVED** [review]
LGTM, only few nits

File: formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt (hunk 2 above)
Comment (Laimiux) [reply formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt:1262]: We have some tests for local only, added a new test for global only.

**Pull request merged**